.github/workflows/build.yml

```diff
@@ -118,6 +121,70 @@ jobs:
           with open(os.environ['GITHUB_OUTPUT'], 'a') as f:
               f.write('\n'.join(f'{key}={value}' for key, value in outputs.items()))
 
+      - name: Build Linux matrix
+        id: linux_matrix
+        env:
+          INPUTS: ${{ toJSON(inputs) }}
+          UPDATE_TO: yt-dlp/yt-dlp@2025.09.05
+        shell: python
+        run: |
+          import json
+          import os
+          EXE_MAP = {
+              'linux': [{
+                  'os': 'linux',
+                  'arch': 'x86_64',
+                  'runner': 'ubuntu-24.04',
+                  'verify_outside_container': True,
+              }, {
+                  'os': 'linux',
+                  'arch': 'aarch64',
+                  'runner': 'ubuntu-24.04-arm',
+                  'verify_outside_container': True,
+              }],
+              'linux_armv7l': [{
+                  'os': 'linux',
+                  'arch': 'armv7l',
+                  'runner': 'ubuntu-24.04-arm',
+                  'qemu_platform': 'linux/arm/v7',
+                  'onefile': False,
+                  'cache_requirements': True,
+              }],
+              'musllinux': [{
+                  'os': 'musllinux',
+                  'arch': 'x86_64',
+                  'runner': 'ubuntu-24.04',
+              }, {
+                  'os': 'musllinux',
+                  'arch': 'aarch64',
+                  'runner': 'ubuntu-24.04-arm',
+              }],
+          }
+          INPUTS = json.loads(os.environ['INPUTS'])
+          matrix = [exe for key, group in EXE_MAP.items() for exe in group if INPUTS.get(key)]
+          skip = bool(matrix)
+          if not matrix:
+              # If we send an empty matrix when no linux inputs are given, "Build artifacts" fails
+              matrix = [EXE_MAP['linux'][0]]
+          for exe in matrix:
+              exe['exe'] = '_'.join(filter(None, (
+                  'yt-dlp',
+                  exe['os'],
+                  exe['arch'] != 'x86_64' and exe['arch'],
+              )))
+              exe.setdefault('qemu_platform', None)
+              exe.setdefault('onefile', True)
+              exe.setdefault('onedir', True)
```

Comment:
should also use these to set `SKIP_ONEFILE_BUILD`, `SKIP_ONEDIR_BUILD`, and `TEST_ONEDIR_BUILD` for docker env